docs/architecture-decisions/.template.md

```diff
@@ -0,0 +1,70 @@
+---
+# Valid statuses: draft | proposed | rejected | accepted | superseded
+status: draft
+author: Your Name
+created: YYYY-MM-DD
+updated: YYYY-MM-DD
+---
+
+# Title
+
+<!--
+This section should be one or two paragraphs that just explains what the goal of this decision is going to be, but without diving too deeply into the "why", "why now", "how", etc.
+Ensure anyone opening the document will form a clear understanding of the intent from reading this paragraph(s).
+-->
+
+## Background
+
+<!--
+The next section is the "Background" section. This section should be at least two paragraphs and can take up to a whole page in some cases.
+The guiding goal of the background section is: as a newcomer to this project (new employee, team transfer), can I read the background section and follow any links to get the full context of why this change is necessary?
+
+If you can't show a random engineer the background section and have them acquire nearly full context on the necessity for the RFC, then the background section is not full enough. To help achieve this, link to prior RFCs, discussions, and more here as necessary to provide context so you don't have to simply repeat yourself.
+-->
+
+## Considered Options
```

Comment:
Can we have a section or two separate sections for motivations and requirements?

This is what confused me the most regarding the evaluation API and the gRPC sync API. Sometimes knowing why we need a feature is more important than how we're gonna implement it.